var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -38,67 +45,184 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
     variant("ocl", default=False, description="Enable OpenCL support")
     variant("tbb", default=False, description="Enable TBB support")
     variant("acc", default=False, description="Enable OpenACC support")
+    variant("hip", default=False, description="Enable HIP support")
     variant("thrust", default=False, description="Enable THRUST support")
     variant("raja", default=False, description="Enable RAJA support")
-    variant("stddata", default=False, description="Enable STD-data support")
-    variant("stdindices", default=False, description="Enable STD-indices support")
-    variant("stdranges", default=False, description="Enable STD-ranges support")
+    variant("std", default=False, description="Enable STD support")
 
     # Some models need to have the programming model abstraction downloaded -
     # this variant enables a path to be provided.
     variant("dir", values=str, default="none", description="Enable Directory support")
+    variant(
+        "sycl2020_submodel",
+        values=("usm", "acc"),
+        when="+sycl2020",
+        default="usm",
+        description="SYCL2020 -> choose between usm and acc methods",
+    )
+    variant(
+        "std_submodel",
+        values=("data", "indices", "ranges"),
+        when="+std",
+        default="data",
+        description="STD -> choose between data, indices and ranges models",
+    )
+    variant(
+        "sycl_opencl_path",
+        values=str,
+        when="+sycl",
+        default="none",
+        description="[ComputeCpp only] Path to OpenCL library, usually called libOpenCL.so",
+    )
 
-    # Kokkos conflict and variant
-    conflicts(
-        "dir=none", when="+kokkos", msg="KOKKKOS requires architecture to be specfied by dir="
+    sycl_compiler_implementations = [
+        "oneapi-icpx",
+        "oneapi-clang",
+        "dpcpp",
+        "hipsycl",
+        "computecpp",
+        "none",
+    ]  # list them here, with the default first
```

Comment:
This is similarly something that should really be modeled as a virtual dependency... I am curious how @becker33 would recommend modeling this one as we transition to compiler deps -- there is already a `sycl` virtual provided by `intel-oneapi-runtime` and `hipsycl`, so that would be my preferred way.

For some of the others I think you should add packages to Spack that provide `sycl`, e.g. `computecpp`. `dpcpp` isn't supported anymore (Intel folks deprecated it) so you might consider leaving it out.

Reply:
ComputeCpp is deprecated so no need for a Spack package.